### PR TITLE
feat: [FCS] - Streamline dependencies on pub sub

### PR DESCRIPTION
**Pull request by PreethiMaai**

### CODE REVIEW

File: cypress/support/cypress-commands/commands.js [hunk 1]
```diff
@@ -892,15 +893,7 @@ Cypress.Commands.add('launchApp', (appType, appCallSign, deviceIdentifier) => {
...
-    data = {
-      query: {
-        params: {
-          [CONSTANTS.APP_ID]: appId,
-          [CONSTANTS.LIFECYCLE_VALIDATION]: true,
-          [CONSTANTS.APP_TYPE]: appCategory,
-        },
-      },
-    };
+    data.query.params[CONSTANTS.LIFECYCLE_VALIDATION] = true;
```
Comment (neeradanelxsi) [inline cypress/support/cypress-commands/commands.js:896]: I believe there was one error when we tried to assign these values directly. something happened (check some bug tickets or check with abhishek)

File: cypress/support/cypress-commands/commands.js [hunk 2]
```diff
@@ -916,8 +909,20 @@ Cypress.Commands.add('launchApp', (appType, appCallSign, deviceIdentifier) => {
...
+
+  if (Cypress.env('additionalLaunchParams')) {
+    const additionalParams = Cypress.env('additionalLaunchParams');
+    for (const key in additionalParams) {
+      let value = additionalParams[key];
+      if (value.startsWith('CYPRESSENV-')) {
+        const envParam = value.split('-')[1];
+        value = getEnvVariable(envParam, false);
+      }
+      data.query.params[key] = value;
```
Comment (neeradanelxsi) [inline cypress/support/cypress-commands/commands.js:925]: add all details in ticket

File: cypress/support/cypress-commands/commands.js (hunk 1 above)
Comment (PreethiMaai) [reply cypress/support/cypress-commands/commands.js]: yes, if the appType is not certification we will error. Updated the code.

**Review by neeradanelxsi — DISMISSED** [review]

File: cypress/support/cypress-support/src/utils.js [hunk 3]
```diff
@@ -205,9 +205,11 @@ function getTopic(
...
   }
   if (operation == CONSTANTS.SUBSCRIBE) {
-    return subscribeSuffix ? topic + subscribeSuffix : topic + CONSTANTS.TOPIC_SUBSCRIBE_SUFFIX;
+    return subscribeSuffix
+      ? topic + subscribeSuffix
+      : topic + getEnvVariable(CONSTANTS.PUB_SUB_SUBSCRIBE_SUFFIX);
   } else {
-    return topic + CONSTANTS.TOPIC_PUBLISH_SUFFIX;
+    return topic + getEnvVariable(CONSTANTS.PUB_SUB_PUBLISH_SUFFIX);
   }
```
Comment (neeradanelxsi) [inline cypress/support/cypress-support/src/utils.js:213]: will this handle if config module does not have these env values ? I believe the ask was to have a default env inside FCS for these ?

File: cypress/support/cypress-commands/commands.js [hunk 4]
```diff
@@ -921,8 +923,30 @@ Cypress.Commands.add('launchApp', (appType, appCallSign, deviceIdentifier) => {
...
+    data.query.params[CONSTANTS.PUB_SUB_UUID] = getEnvVariable(CONSTANTS.PUB_SUB_UUID);
+  }
+  if (getEnvVariable(CONSTANTS.PUB_SUB_PUBLISH_SUFFIX, false)) {
+    data.query.params[CONSTANTS.PUB_SUB_PUBLISH_SUFFIX] = getEnvVariable(
+      CONSTANTS.PUB_SUB_SUBSCRIBE_SUFFIX
+    );
+  }
+  if (getEnvVariable(CONSTANTS.PUB_SUB_SUBSCRIBE_SUFFIX, false)) {
+    data.query.params[CONSTANTS.PUB_SUB_SUBSCRIBE_SUFFIX] = getEnvVariable(
+      CONSTANTS.PUB_SUB_PUBLISH_SUFFIX
```
Comment (neeradanelxsi) [inline cypress/support/cypress-commands/commands.js:937]: comments are missing

File: cypress/support/cypress-commands/commands.js [hunk 5]
```diff
@@ -921,8 +923,30 @@ Cypress.Commands.add('launchApp', (appType, appCallSign, deviceIdentifier) => {
...
+    );
+  }
+
+  if (Cypress.env('additionalLaunchParams')) {
```
Comment (neeradanelxsi) [inline cypress/support/cypress-commands/commands.js:941]: suppose a user wants to pass additional parameters in additionalLaunchParams , where is the documentation for that in FCS ?

File: cypress/support/cypress-commands/commands.js [hunk 6]
```diff
@@ -921,8 +923,30 @@ Cypress.Commands.add('launchApp', (appType, appCallSign, deviceIdentifier) => {
...
+  }
+  if (getEnvVariable(CONSTANTS.PUB_SUB_UUID, false)) {
+    data.query.params[CONSTANTS.PUB_SUB_UUID] = getEnvVariable(CONSTANTS.PUB_SUB_UUID);
+  }
+  if (getEnvVariable(CONSTANTS.PUB_SUB_PUBLISH_SUFFIX, false)) {
+    data.query.params[CONSTANTS.PUB_SUB_PUBLISH_SUFFIX] = getEnvVariable(
+      CONSTANTS.PUB_SUB_SUBSCRIBE_SUFFIX
+    );
+  }
+  if (getEnvVariable(CONSTANTS.PUB_SUB_SUBSCRIBE_SUFFIX, false)) {
```
[the remark this answers is not in the document]
Comment (PreethiMaai) [reply cypress/support/cypress-commands/commands.js:935]: Updated the condition
image

File: cypress/support/cypress-support/src/utils.js (hunk 3 above)
Comment (PreethiMaai) [reply cypress/support/cypress-support/src/utils.js:213]: yes, added default env inside cypress.config.js. Apologies for not adding the changes in the PR before.

File: cypress/support/cypress-commands/commands.js (hunk 4 above)
Comment (PreethiMaai) [reply cypress/support/cypress-commands/commands.js:940]: added

File: cypress/support/cypress-commands/commands.js (hunk 5 above)
Comment (PreethiMaai) [reply cypress/support/cypress-commands/commands.js:945]: added

Comment (neeradanelxsi) [reply cypress/support/cypress-commands/commands.js:945]: Not in comments. It must be added in readme

Comment (neeradanelxsi) [reply cypress/support/cypress-commands/commands.js:945]: It must be written in such a way that after reading the readme document, a user is able to add a launch parameter (like testtoken  , url etc) .

File: cypress.config.js [hunk 7]
```diff
@@ -87,6 +87,8 @@ const env = {
   ],
   fcaAppList: ['default3rdPartyAppId'],
   variableObjectsPrefixLists: ['DYNAMIC_FB_CALL_VARIABLES'],
+  pubSubPublishSuffix: '_FCS',
+  pubSubSubscribeSuffix: '_FCA'
 };
```
[the remark this answers is not in the document]
Comment (PreethiMaai) [reply cypress.config.js:92]: included

File: cypress/support/cypress-commands/commands.js (hunk 5 above)
Comment (PreethiMaai) [reply cypress/support/cypress-commands/commands.js:945]: added readme

File: cypress/support/cypress-commands/commands.js [hunk 8]
```diff
@@ -921,8 +923,36 @@ Cypress.Commands.add('launchApp', (appType, appCallSign, deviceIdentifier) => {
...
+  // If a key exists in both the default parameters and the additional parameters, the value from the additional parameters will override the default value.
+  if (Cypress.env('additionalLaunchParams')) {
+    const additionalParams = Cypress.env('additionalLaunchParams');
+    for (const key in additionalParams) {
+      let value = additionalParams[key];
+      // If the value starts with 'CYPRESSENV-', extract the variable name.
+      if (value.startsWith('CYPRESSENV-')) {
+        const envParam = value.split('-')[1];
+        // Fetch the corresponding value from the environment.
+        value = getEnvVariable(envParam, false);
```
Comment (jbigel) [inline cypress/support/cypress-commands/commands.js:953]: If we fail to get the variable should we still add to the query params?

Comment (PreethiMaai) [reply cypress/support/cypress-commands/commands.js:953]: updated condition to ensure we don't add variables to the query parameters if we fail to retrieve.

**Review by neeradanelxsi — APPROVED** [review]

**Review by jmeier204 — APPROVED** [review]

**Review by jbigel — APPROVED** [review]

**Pull request merged**